selenium/exercise_12.rb

```diff
@@ -0,0 +1,109 @@
+require 'selenium-webdriver'
+
+class Exercise_12
+    attr_accessor :driver, :site, :wait
+
+    def initialize(path_to_driver, url)
+        @wait = Selenium::WebDriver::Wait.new(:timeout => 20)
+        Selenium::WebDriver::Chrome.driver_path = path_to_driver
+        @driver = Selenium::WebDriver.for :chrome
+        @site_url = url
+        @driver.manage.window.maximize
+        @driver.get(@site_url)
+    end
+    
+    def login()
+        email = "madhusamudra77@gmail.com"
+        password = "stackoverflow11"
+
+        wait.until { driver.find_element(:css, "nav ol li:nth-of-type(3) a") }
+        driver.find_element(:css, "nav ol li:nth-of-type(3) a").click
+        sleep(2)
+        wait.until { driver.find_element(:id, "formContainer") }
+        driver.find_element(:id, "email").send_keys(email)
+        driver.find_element(:id, "password").send_keys(password)
+        driver.find_element(:id, "submit-button").click
+        sleep(2)
+    end
+
+    def search()
+        if driver.find_element(:class, "js-accept-cookies").displayed?
+            driver.find_element(:class, "js-accept-cookies").click
+        end
+        sleep(3)
+
+        wait.until { driver.find_element(:name, "q") }
+        search_bar = driver.find_element(:name, "q")
+        search_bar.send_keys("ruby")
+        sleep(2)
+        search_bar.send_keys(:enter)
+        driver.navigate.to "https://stackoverflow.com/questions/tagged/ruby"
```

Comment:
This is because on pressing enter key, it shows a page where it asks for verification via recaptcha. Thus I used <driver.navigate.to "https://stackoverflow.com/questions/tagged/ruby"> to avoid accessing the captcha page